### PR TITLE
kb(grid): Add KB for changing the search results on column hide or show

### PR DESCRIPTION
**Pull request by dimodi**

### CODE REVIEW

File: knowledge-base/grid-search-column-menu.md [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    </tbody>
+</table>
+
+## Description
+
+This KB article answers the following questions:
+
+* How to change the Grid search results when the user hides or shows a string column?
+* How to refresh the Grid search results when the `Visible` parameter of a column changes?
+* How to adjust and match the Grid search results to the columns with a `Visible` attribute that changed runtime?
```
Comment (dimodi) [inline knowledge-base/grid-search-column-menu.md:29]:
```suggestion
* How to refresh and match the Grid search results when the `Visible` attribute of a column changes?
```

File: knowledge-base/grid-search-column-menu.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+This KB article answers the following questions:
+
+* How to change the Grid search results when the user hides or shows a string column?
+* How to refresh and match the Grid search results when the `Visible` attribute of a column changes?
+* How to hide or show search results in the Grid when the visibility of a column changes and a column is no longer displayed?
+
+## Solution
+
+1. Subscribe to the [Grid `OnStateChanged` event]({%slug grid-state%}#onstatechanged).
+1. Check if the `PropertyName` event argument is equal to `"ColumnStates"` to verify that the user modified the column state.
```
Comment (ikoevska) [inline knowledge-base/grid-search-column-menu.md:34]: I know that our style guide advocates for using simple tenses. However, in this case the perfect tense fits better grammatically.

```suggestion
1. Check if the `PropertyName` event argument is equal to `"ColumnStates"` to verify that the user has modified the column state.
```

File: knowledge-base/grid-search-column-menu.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+1. Check for `FilterDescriptor` instances in `args.GridState.SearchFilter.FilterDescriptors` to verify if a search is active.
+1. [Get the visible columns from `args.GridState.ColumnStates`]({%slug grid-kb-column-state%}). Use only the columns with a `Field` that points to a `string` property.
```
Comment (ikoevska) [inline knowledge-base/grid-search-column-menu.md:36]: When the link is a complete sentence, it's best to include the ending punctuation as well.

```suggestion
1. [Get the visible columns from `args.GridState.ColumnStates`.]({%slug grid-kb-column-state%}) Use only the columns with a `Field` that points to a `string` property.
```

File: knowledge-base/grid-search-column-menu.md [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+                            Member = field,
+                            MemberType = typeof(string),
+                            Operator = FilterOperator.Contains,
+                            Value = searchValue
+                        });
+                        shouldRebindGrid = true;
+                    }
+                }
+
+                // Apply the changes in args.GridState.SearchFilter and rebind the Grid.
```
Comment (ikoevska) [inline knowledge-base/grid-search-column-menu.md:130]: Note how this code comment has ending punctuation while other code comments don't. Ideally, the use of ending punctuation should be consistent (I don't have strict preferences in either direction).

File: knowledge-base/grid-search-column-menu.md [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+        public int Id { get; set; }
+        public string Name { get; set; } = string.Empty;
+        public string Description { get; set; } = string.Empty;
+    }
+}
+````
+
+## See Also
+
+* [Search in hidden Grid columns]({%slug grid-kb-search-in-hidden-fields%})
```
Comment (ikoevska) [inline knowledge-base/grid-search-column-menu.md:166]:
```suggestion
* [Search in Hidden Grid Columns]({%slug grid-kb-search-in-hidden-fields%})
```

**Review by ikoevska — APPROVED** [review]

**Pull request merged**